server/@types/ui/index.d.ts

```diff
@@ -0,0 +1,130 @@
+import { RoshRisks, RiskTier, FlagsEnvelope, Mappa } from '@approved-premises-shared'
+
+// A utility type that allows us to define an object with a date attribute split into
+// date, month, year (and optionally, time) attributes. Designed for use with the GOV.UK
+// date input
+export type ObjectWithDateParts<K extends string | number> = { [P in `${K}-${'year' | 'month' | 'day'}`]: string } & {
+  [P in `${K}-time`]?: string
+} & {
+  [P in K]?: string
+}
+
+export type BookingStatus = 'arrived' | 'awaiting-arrival' | 'not-arrived' | 'departed' | 'cancelled'
+
+export type TaskNames = 'basic-information' | 'type-of-ap'
+
+export interface HtmlAttributes {
+  [key: string]: string
+}
+
+export interface TextItem {
+  text: string
+}
+
+export interface HtmlItem {
+  html: string
+}
+
+export type TableCell = { text: string; attributes?: HtmlAttributes } | { html: string }
+export interface TableRow {
+  [index: number]: TableCell
+}
+
+export interface RadioItem {
+  text: string
+  value: string
+  checked?: boolean
+}
+
+export interface CheckBoxItem {
+  text: string
+  value: string
+  checked?: boolean
+}
+
+export interface SelectOption {
+  text: string
+  value: string
+  selected?: boolean
+}
+
+export interface SummaryList {
+  classes?: string
+  attributes?: HtmlAttributes
+  rows: Array<SummaryListItem>
+}
+
+export interface SummaryListActionItem {
+  href: string
+  text: string
+  visuallyHiddenText: string
+}
+
+export interface SummaryListItem {
+  key: TextItem | HtmlItem
+  value: TextItem | HtmlItem
+  actions?: { items: Array<SummaryListActionItem> }
+}
+
+export interface IdentityBarMenu {
+  items: Array<IdentityBarMenuItem>
+}
+
+export interface IdentityBarMenuItem {
+  classes: string
+  href: string
+  text: string
+}
+
+export type RiskLevel = 'Low' | 'Medium' | 'High' | 'Very High'
+
+export type RiskEnvelopeStatus = 'retrieved' | 'not_found' | 'error'
+
+export type TierNumber = '1' | '2' | '3' | '4'
+export type TierLetter = 'A' | 'B' | 'C' | 'D'
+export type RiskTierLevel = `${TierLetter}${TierNumber}`
```

Comment:
Should these be in the models, or is the generator not smart enough to do this? If it's the latter, do they actually get used anywhere?

Reply:
The RiskTierLevel and RiskLevel are currently used in the applicationSummary factory (soon to be RIP) and the risks factory. 
RiskEnvlopeStatus is a dupe of a type in the model so I'll remove

I think that the generator is only as smart as what we tell it - I think we just have `string` for the risk tier level.